__tests__/integration/stream.test.ts

```diff
@@ -320,15 +346,53 @@ describe("StreamClient", () => {
     stream.start(
       function onEvent(_) {},
       function onError(e) {
-        // TODO: be more specific about the error and split into multiple tests
-        expect(e).toBeInstanceOf(ServiceError);
+        if (e instanceof AbortError) {
+          expect(e.httpStatus).toBe(400);
+          expect(e.abort).toBe("oops");
+        }
+        resolve();
+      }
+    );
+
+    await promise;
+  });
+
+  it("handles a QueryRuntimeError via callback when processing an event", async () => {
+    expect.assertions(1);
+
+    const response = await client.query<StreamToken>(
+      fql`StreamTest.all().map((doc) => notARealFn(doc)).toStream()`
+    );
+    const token = response.data;
+
+    const stream = new StreamClient<DocumentT<StreamTest>>(
+      token,
+      defaultStreamConfig
+    );
+
+    // create some events that will be played back
+    await client.query(fql`StreamTest.create({ value: 0 })`);
+
+    let resolve: () => void;
+    const promise = new Promise((res) => {
+      resolve = () => res(null);
+    });
+
+    stream.start(
+      function onEvent(_) {},
+      function onError(e) {
+        if (e instanceof QueryRuntimeError) {
+          expect(e.httpStatus).toBe(400);
+        }
         resolve();
       }
     );
 
     await promise;
   });
 
+  // TODO: test other error events
```

Comment:
is this something we can just do now?

Reply:
If you can think of ways to trigger them. I know in theory you could, for example, get an error due to throughput limits, but I'm not sure how to reliably test that.

Reply:
Ah i see.

Mocking would be the best bet for those.

Reply:
Try the 2nd approach then with the verificaiton and associated unit test.